### PR TITLE
Improve script to install Ceph AIO without Vagrant

### PR DESCRIPTION
**Pull request by winggundamth**
- Add mon_pg_warn_max_per_osd, osd_pool_default_size and osd_pool_default_min_size variables
- Add option to configure default pool size
- Add option not to install MDS and RGW
- Improve ssh_setup to check if file and key already exist
- Improve populate_vars for safety rerun for idempotent
- Change path of inventory file to where ansible run. So this won't pollute the system
- Remove check host key
- Remove sudo when install Ansible

### CODE REVIEW

File: roles/ceph-common/defaults/main.yml [hunk 1]
```diff
@@ -250,6 +250,7 @@ monitor_interface: interface
 monitor_address: 0.0.0.0
 mon_use_fqdn: false # if set to true, the MON name used will be the fqdn in the ceph.conf
 monitor_address_block: false
+mon_pg_warn_max_per_osd: 300
```
Comment (leseb) [inline roles/ceph-common/defaults/main.yml]: No no no, for this you have to use `ceph_conf_overrides` :), see https://github.com/ceph/ceph-ansible/blob/master/group_vars/all.yml.sample#L313-L328

File: roles/ceph-common/defaults/main.yml [hunk 2]
```diff
@@ -260,6 +261,8 @@ osd_mkfs_type: xfs
 osd_mkfs_options_xfs: -f -i size=2048
 osd_mount_options_xfs: noatime,largeio,inode64,swalloc
 osd_objectstore: filestore
+osd_pool_default_size: 3
```
Comment (leseb) [inline roles/ceph-common/defaults/main.yml]: ditto

File: roles/ceph-common/defaults/main.yml [hunk 3]
```diff
@@ -260,6 +261,8 @@ osd_mkfs_type: xfs
...
+osd_pool_default_min_size: 2
```
Comment (leseb) [inline roles/ceph-common/defaults/main.yml]: ditto

File: roles/ceph-common/templates/ceph.conf.j2 [hunk 4]
```diff
@@ -12,9 +12,12 @@ auth supported = none
 fsid = {{ fsid }}
 {% endif %}
 max open files = {{ max_open_files }}
-{% if common_single_host_mode is defined %}
+{% if common_single_host_mode is defined and common_single_host_mode %}
 osd crush chooseleaf type = 0
 {% endif %}
+mon pg warn max per osd = {{ mon_pg_warn_max_per_osd }}
```
Comment (leseb) [inline roles/ceph-common/templates/ceph.conf.j2]: not needed as per my comment above

File: ceph-aio-no-vagrant.sh [hunk 5]
```diff
@@ -95,37 +114,53 @@ function cp_var {
...
 function create_inventory {
-  cat > /etc/ansible/hosts <<EOF
+  cat > hosts <<EOF
 [mons]
 localhost
 [osds]
 localhost
+EOF
+  if [ "$INSTALL_MDS" = true ] ; then
+    cat >> hosts <<EOF
```
Comment (leseb) [inline ceph-aio-no-vagrant.sh]: I don't think this is idempotent, in the condition can you also check for an existing section? Therefore we don't add this section N times while re-running the script.

File: ceph-aio-no-vagrant.sh [hunk 6]
```diff
@@ -95,37 +114,53 @@ function cp_var {
...
 [mdss]
 localhost
+EOF
+  fi
+  if [ "$INSTALL_RGW" = true ] ; then
```
Comment (leseb) [inline ceph-aio-no-vagrant.sh]: ditto

File: ceph-aio-no-vagrant.sh (hunk 5 above)
Comment (leseb) [inline ceph-aio-no-vagrant.sh]: Right I missed that.

File: ceph-aio-no-vagrant.sh [hunk 7]
```diff
@@ -95,37 +114,57 @@ function cp_var {
...
+  sed -i "s/[#]*monitor_address: .*/monitor_address: ${IP}/" group_vars/all.yml
+  sed -i "s/[#]*journal_size: .*/journal_size: 100/" group_vars/all.yml
+  sed -i "s|[#]*public_network: .*|public_network: ${SUBNET}|" group_vars/all.yml
+  sed -i "s/[#]*common_single_host_mode: .*/common_single_host_mode: true/" group_vars/all.yml
+
+  if ! grep -q '^ceph_conf_overrides:' group_vars/all.yml; then
+    cat >> group_vars/all.yml <<EOF
+ceph_conf_overrides:
+  global:
+    mon pg warn max per osd: 0
```
Comment (leseb) [inline ceph-aio-no-vagrant.sh]: Spaces won't work here so you need to:

s/mon pg warn max per osd/mon_pg_warn_max_per_osd/

Comment (leseb) [inline ceph-aio-no-vagrant.sh]: It's working for me as well :)

File: ceph-aio-no-vagrant.sh [hunk 8]
```diff
@@ -95,37 +114,57 @@ function cp_var {
...
+    osd pool default size: 2
+    osd pool default min size: 2
+EOF
+  fi
+  sed -i "s/^    osd pool default size: .*/    osd pool default size: $CEPH_POOL_DEFAULT_SIZE/" group_vars/all.yml
```
Comment (leseb) [inline ceph-aio-no-vagrant.sh]: Since we use `ceph_conf_overrides` we can remove this right?

Comment (winggundamth) [inline ceph-aio-no-vagrant.sh]: No. This is where I reconfigure pool default size variables as it specify with -p parameter.

**Comment by leseb** [comment]
Thanks @winggundamth

**Comment by winggundamth** [comment]
My first pull request on ceph-ansible project. Thanks @leseb for helping me too :+1:

**Pull request merged**